test/e2e/app/app.go

```diff
@@ -376,12 +392,15 @@ func (app *Application) ProcessProposal(_ context.Context, req *abci.RequestProc
 			return &abci.ResponseProcessProposal{Status: abci.ResponseProcessProposal_REJECT}, nil
 		}
 		// Additional check for vote extension-related txs
-		if k == voteExtensionKey {
-			_, err := strconv.Atoi(v)
-			if err != nil {
-				app.logger.Error("malformed vote extension transaction", k, v, "err", err)
+		switch {
+		case k == voteExtensionKey:
+			if err := app.verifyExtensionTx(req.Height, v); err != nil {
+				app.logger.Error("vote extension transaction failed verification, rejecting proposal", k, v, "err", err)
```

Comment:
Add a comment regarding using this proposal as evidence of misbehavior by the proposer?

Reply:
Added a NOTE